static/solo/js/menu.js

```diff
@@ -0,0 +1,225 @@
+            menuFrag.appendChild(slideMenu);
+            _frame[0].parentNode.insertBefore(menuFrag, _frame[0].nextSibling);
+        }
+    };
+
+    this.listenToResize = function(argument) {
+        w.addEventListener('resize',function(){
+            setTimeout(function(){
+                var newSize = (that.payerCont.clientHeight - 40) +'px;';
+                that.hiddenMenu.style.cssText = 'height:'+newSize;
+            },50)
+        },false);
+    };
+
+    this.update = function(newData) {
+
+      if (that.noVidDiv) {
+        that.deleteDivs();
+        for (var i = 0; i < newData.length; i++) {
+            that.allVideos.push(newData[i]);
+            settings.data.push(newData[i]);
+            that.noVidDiv.setAttribute('id', 'tvp-clearfix');
+            that.noVidDiv.innerHTML += Utils.tmpl(itemTemplate, newData[i]);
+            that.scrollMenu.appendChild(that.noVidDiv);
+        }
+      }else{
+        var newVivFrag = d.createDocumentFragment(),
+            newDiv = d.createElement('div');
+        for (var i = 0; i < newData.length; i++) {
+            that.allVideos.push(newData[i]);
+            settings.data.push(newData[i]);
+            newDiv.setAttribute('id', 'tvp-clearfix');
+            newDiv.innerHTML += Utils.tmpl(itemTemplate, newData[i]);
+            newVivFrag.appendChild(newDiv);
+
+            that.scrollMenu.appendChild(newDiv);
+        }
+      }
+      that.cacheDOM();
+      that.bindClickEvent();
+    };
+
+    this.deleteDivs = function(){
+        for (var i = that.tvpNoVids.length - 1; i >= 0; i--){
+            that.noVidDiv.removeChild(that.tvpNoVids[i]);
+        }
+    };
+
+    this.clearActiveItems = function () {
+        for (var i = that.tvpVid.length - 1; i >= 0; i--) {
+            if(that.tvpVid[i].classList.contains('active')){
+                that.tvpVid[i].classList.remove('active');
+            }
+        }
+    };
+
+    this.setActiveItem = function (id) {
+        for (var i = that.tvpVid.length - 1; i >= 0; i--) {
+            var item = that.tvpVid[i],
+                itemId = item.id.split('-').pop();
+            if (itemId === id && !item.classList.contains('active')){
+                this.clearActiveItems();
+                item.classList.add('active');
+            }
+        }
+    };
+
+    this.videoClick = function(vids){
+        vids.onclick = function() {
+            if (!this.classList.contains('tvp-video')) return;
+
+            that.clearActiveItems();
+            this.classList.add('active');
+
+            var id = this.id.split('-').pop(),
+                selected = that.allVideos.filter(function(v){return v.id === id});
+            player.play(player.createAsset(selected[0]));
+            that.toggleMenu();
+        };
+    };
+    player.onNext = function(){
+        var playerAsset = player.assets[player.current];
+        that.setActiveItem(playerAsset.assetId);
+        that.hideMenu();
+    };
+    player.isReady = function(){
```

Comment:
You should not create functions to the dependency n the fly, this has to be passed as options to the dependency constructor, for instance:

```
playerSettings.onReady = function(){
  menu.init();
};

playerSettings.onNext = function(nextAsset){
  menu.setActiveItem(nextAsset.assetId);
  menu.hideMenu();
};

playerSettings.onFullscreenChange = function(){
  menu.hideMenu();
  menu.hiddenMenu.style.cssText = (window.innerHeight - 40) +'px;';
};
```